x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CloudSetup } from '@kbn/cloud-plugin/server';
+import { DeprecationsDetails } from '@kbn/core-deprecations-common';
+import { GetDeprecationsContext, RegisterDeprecationsConfig } from '@kbn/core-deprecations-server';
+
+import { hasFleetServers } from '@kbn/fleet-plugin/server/services/fleet_server';
+import { isAgentlessEnabled } from '@kbn/fleet-plugin/server/services/utils/agentless';
+import { i18n } from '@kbn/i18n';
+import { Connector, fetchConnectors } from '@kbn/search-connectors';
+
+import { ConfigType } from '..';
+
+export const getRegisteredDeprecations = (
+  config: ConfigType,
+  cloud: CloudSetup,
+  docsUrl: string
+): RegisterDeprecationsConfig => {
+  return {
+    getDeprecations: async (ctx: GetDeprecationsContext) => {
+      const hasAgentless = isAgentlessEnabled();
+      const hasFleetServer = await hasFleetServers(
+        ctx.esClient.asInternalUser,
+        ctx.savedObjectsClient
+      );
+      return [
+        ...getEnterpriseSearchNodeDeprecation(config, cloud, docsUrl),
+        ...(await getCrawlerDeprecations(ctx, docsUrl)),
+        ...(await getNativeConnectorDeprecations(
+          ctx,
+          hasAgentless,
+          hasFleetServer,
+          cloud,
+          docsUrl
+        )),
+      ];
+    },
+  };
+};
+
+/**
+ * If Enterprise Search Node is configured, it's marked as a critical deprecation
+ * Warns that removing the node will disable crawlers/connectors
+ */
+export function getEnterpriseSearchNodeDeprecation(
+  config: ConfigType,
+  cloud: CloudSetup,
+  docsUrl: string
+): DeprecationsDetails[] {
+  if (config.host) {
+    const steps = [];
+    let addendum: string = '';
+    const isCloud = !!cloud?.cloudId;
+    if (isCloud) {
+      steps.push(
+        ...[
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.gotocloud', {
+            values: { baseUrl: cloud.baseUrl },
+            defaultMessage: 'Go to {baseUrl}',
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clickedit', {
+            defaultMessage: "Click the 'Edit' tab",
```

Comment:
@seanstory once on https://cloud.elastic.co, shouldn't the next step be to select the right deployment to manage?

What do you think about using smth like 
1. [Click here to manage your deployment](${cloud.baseUrl + '/deployments/' + cloud.deploymentId}) **or** go to {baseUrl} and click "Manage" for the deployment you're upgrading.
2. Click the 'Edit' tab
3. ..... and so on ...

Reply:
> [Click here to manage your deployment](${cloud.baseUrl + '/deployments/' + cloud.deploymentId}) or go to {baseUrl} and click "Manage" for the deployment you're upgrading.

Unfortunately, I can't do hyperlinks in the `steps`. So I could give the raw string to their deployment (which they'd have to copy paste), or I could say `"Click the link above to go to your deployment"`, or I could give a step 2 of "choose this deployment" after "go to cloud.elastic.co`"

But you're right, this is missing a step to get them on a single deployment.

Reply:
how do you feel about: https://github.com/elastic/kibana/pull/205500/commits/76455e692e50cd83c2e02a8adc31b2c2a347a51f ?